contracts/contracts/validator-registry/VanillaRegistry.sol

```diff
@@ -128,7 +128,8 @@ contract VanillaRegistry is IVanillaRegistry, VanillaRegistryStorage,
      * @param blsPubKeys The BLS public keys to add stake to.
      */
     function addStake(bytes[] calldata blsPubKeys) external payable 
-        onlyExistentValidatorRecords(blsPubKeys) onlyNotUnstaking(blsPubKeys) whenNotPaused() {
+        onlyExistentValidatorRecords(blsPubKeys) onlyWithdrawalAddress(blsPubKeys)
```

Comment:
Added `onlyWithdrawalAddress` modifier here s.t. only the withdrawal address of the existing validator record(s) can add stake